src/OpenSearch/Endpoints/AbstractEndpoint.php

```diff
@@ -280,4 +282,39 @@ private function isNestedArray(array $a): bool
 
         return false;
     }
+
+    /**
+     * This function returns all param deprecations also optional with an replacement field
+     *
+     * @return array<string, string|null>
+     */
+    protected function getParamDeprecation(): array
+    {
+        return [];
+    }
+
+    private function checkForDeprecations(): void
+    {
+        $deprecations = $this->getParamDeprecation();
+
+        if ($deprecations === []) {
+            return;
+        }
+
+        $keys = array_keys($this->params);
+
+        foreach ($keys as $key) {
+            if (array_key_exists($key, $deprecations)) {
+                $val = $deprecations[$key];
+
+                $msg = sprintf('The parameter "%s" is deprecated and will be removed without replacement', $key);
```

Comment:
Can we also add that stating that it will be removed `in the next major version`?